lib/model/katex.dart

```diff
@@ -0,0 +1,566 @@
+          // .mathbb,
+          // .textbb { font-family: KaTeX_AMS; }
+          styles.fontFamily = 'KaTeX_AMS';
+          classFound = true;
+
+        case 'mathcal':
+          // .mathcal { font-family: KaTeX_Caligraphic; }
+          styles.fontFamily = 'KaTeX_Caligraphic';
+          classFound = true;
+
+        case 'mathfrak':
+        case 'textfrak':
+          // .mathfrak,
+          // .textfrak { font-family: KaTeX_Fraktur; }
+          styles.fontFamily = 'KaTeX_Fraktur';
+          classFound = true;
+
+        case 'mathboldfrak':
+        case 'textboldfrak':
+          // .mathboldfrak,
+          // .textboldfrak { font-family: KaTeX_Fraktur; font-weight: bold; }
+          styles.fontFamily = 'KaTeX_Fraktur';
+          styles.fontWeight = KatexSpanFontWeight.bold;
+          classFound = true;
+
+        case 'mathtt':
+          // .mathtt { font-family: KaTeX_Typewriter; }
+          styles.fontFamily = 'KaTeX_Typewriter';
+          classFound = true;
+
+        case 'mathscr':
+        case 'textscr':
+          // .mathscr,
+          // .textscr { font-family: KaTeX_Script; }
+          styles.fontFamily = 'KaTeX_Script';
+          classFound = true;
+      }
+
+      // We can't add the case for the next class (.mathsf, .textsf) in the
+      // above switch block, because there is already a case for .textsf above.
+      // So start a new block, to keep the order of the cases here same as the
+      // CSS class definitions in katex.scss .
+      switch (spanClass) {
+        case 'mathsf':
+        case 'textsf':
+          // .mathsf,
+          // .textsf { font-family: KaTeX_SansSerif; }
+          styles.fontFamily = 'KaTeX_SansSerif';
+          classFound = true;
+
+        case 'mathboldsf':
+        case 'textboldsf':
+          // .mathboldsf,
+          // .textboldsf { font-family: KaTeX_SansSerif; font-weight: bold; }
+          styles.fontFamily = 'KaTeX_SansSerif';
+          styles.fontWeight = KatexSpanFontWeight.bold;
+          classFound = true;
+
+        case 'mathsfit':
+        case 'mathitsf':
+        case 'textitsf':
+          // .mathsfit,
+          // .mathitsf,
+          // .textitsf { font-family: KaTeX_SansSerif; font-style: italic; }
+          styles.fontFamily = 'KaTeX_SansSerif';
+          styles.fontStyle = KatexSpanFontStyle.italic;
+          classFound = true;
+
+        case 'mainrm':
+          // .mainrm { font-family: KaTeX_Main; font-style: normal; }
+          styles.fontFamily = 'KaTeX_Main';
+          styles.fontStyle = KatexSpanFontStyle.normal;
+          classFound = true;
+
+        // TODO handle skipped class declarations between .mainrm and
+        //   .sizing .
+
+        case 'sizing':
+        case 'fontsize-ensurer':
+          // .sizing,
+          // .fontsize-ensurer { ... }
+          if (index + 2 < spanClass.length) {
+            final resetSizeClass = spanClasses[index + 1];
+            final sizeClass = spanClasses[index + 2];
```

Comment:
Oh, one bug here:
```suggestion
          if (index + 2 < spanClasses.length) {
            final resetSizeClass = spanClasses[index + 1];
            final sizeClass = spanClasses[index + 2];
```

:slightly_smiling_face: